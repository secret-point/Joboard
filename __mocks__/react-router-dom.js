// Mock navigate and useParams when needed
// const navigate = jest.fn();
// const useParams = jest.fn();
const useLocation = jest.fn();

module.exports = {
  ...jest.requireActual("react-router-dom"),
  // useNavigate: () => navigate,
  // useParams,
  useLocation,
};
