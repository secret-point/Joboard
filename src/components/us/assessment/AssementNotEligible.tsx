import React, { useEffect } from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text, H4 } from "@amzn/stencil-react-components/text";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { checkAndBoundGetApplication, getLocale, goToCandidateDashboard } from "../../../utils/helper";
import { connect } from "react-redux";
import { ApplicationState } from "../../../reducers/application.reducer";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { useLocation } from "react-router";
import { getPageNameFromPath, parseQueryParamsArrayToSingleItem } from "../../../helpers/utils";
import queryString from "query-string";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { JobState } from "../../../reducers/job.reducer";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { translate as t } from "../../../utils/translator";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";

interface MapStateToProps {
  application: ApplicationState,
  candidate: CandidateState,
  job: JobState
}

interface AssessmentNotEligibleProps {

}

type AssessmentNotEligibleMergedProps = MapStateToProps & AssessmentNotEligibleProps;

export const AssessmentNotEligible = (props: AssessmentNotEligibleMergedProps) => {

  const { application, job, candidate } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId } = queryParams;
  const applicationData = application.results;
  const jobDetail = job.results;
  const candidateData = candidate.results.candidateData;

  useEffect(() => {
    boundGetCandidateInfo();
  },[])

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  // Don't refresh data if id is not changing
  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() })
  }, [jobDetail, jobId]);

  useEffect(() => {
    jobDetail && applicationData && addMetricForPageLoad(pageName);
  }, [jobDetail, applicationData, pageName]);

  return (
    <Col gridGap={15}>
      <H4>
        {t("BB-assessment-not-eligible-heading-text", "{name}, your interest with Amazon means a lot to us.", {name: candidateData?.firstName || ''})}
      </H4>
      <Text fontSize="T200">
        {t("BB-assessment-not-eligible-thank-you-and-assessment-not-eligible-notice-text", "Thank you so much for giving us the opportunity to consider you for this position. Sadly, we will not be moving forward with your candidacy at this time.")}
      </Text>
      <Text fontSize="T200">
        {t("BB-assessment-not-eligible-reconsideration-notice-text", "To be reconsidered for this role at another time, you may take the assessment again in three months.")}
      </Text>
      <Col padding={{top: 'S300'}}>
        <Button
          variant={ButtonVariant.Primary}
          onClick={goToCandidateDashboard}
        >
          {t("BB-assessment-not-eligible-back-to-dashboard-button-text", "Return to dashboard")}
        </Button>
      </Col>
    </Col>
  )
}

const mapStateToProps = ( state: MapStateToProps ) => {
  return state;
};

export default connect(mapStateToProps)(AssessmentNotEligible);
