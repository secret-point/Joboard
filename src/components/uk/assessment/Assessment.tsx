import React, { useEffect } from "react";
import { Col, } from "@amzn/stencil-react-components/layout";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { JobState } from "../../../reducers/job.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { checkAndBoundGetApplication, getLocale } from "../../../utils/helper";
import StepHeader from "../../common/StepHeader";
import { ApplicationStepListUK } from "../../../utils/constants/common";
import { ApplicationState } from "../../../reducers/application.reducer";
import IFrame from "../../common/IFrame";
import { getStepsByTitle } from "../../../helpers/steps-helper";
import { APPLICATION_STEPS as STEPS } from "../../../utils/enums/common";

interface MapStateToProps {
  job: JobState;
  schedule: ScheduleState;
  application: ApplicationState;
}

export const Assessment = (props: MapStateToProps) => {
  const { job, schedule, application } = props;
  const { search, pathname } = useLocation();
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { jobId, applicationId } = queryParams;
  const jobDetail = job.results;
  const { scheduleId } = queryParams;
  const { scheduleDetail } = schedule.results;
  const pageName = getPageNameFromPath(pathname);
  const { assessment } = application?.results || {};
  const headerStep = getStepsByTitle(ApplicationStepListUK, STEPS.COMPLETE_AN_ASSESSMENT)[0]; 

  // Don't refetch data if id is not changing
  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobId]);

  useEffect(() => {
    scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    });
  }, [scheduleId]);
  
  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    if (jobDetail && ((scheduleId && scheduleDetail) || (!scheduleId))) {
      addMetricForPageLoad(pageName);
    }
  }, [jobDetail, scheduleDetail]);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, []);

  return (
    <>
      <Col gridGap="S300" padding="0">
        <StepHeader jobTitle={jobDetail?.jobTitle || ""} step={headerStep} steps={ApplicationStepListUK} />
        {assessment?.assessmentUrl && <IFrame src={ assessment.assessmentUrl} />}
      </Col>
    </>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(Assessment);
