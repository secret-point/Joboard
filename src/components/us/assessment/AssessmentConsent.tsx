import React, { useEffect, useState } from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { H4, Text } from "@amzn/stencil-react-components/text";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { checkAndBoundGetApplication, getLocale, onAssessmentStart } from "../../../utils/helper";
import { ApplicationState } from "../../../reducers/application.reducer";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import queryString from "query-string";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { JobState } from "../../../reducers/job.reducer";
import { translate as t } from "../../../utils/translator";
import InnerHTML from 'dangerously-set-html-content';
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";

interface MapStateToProps {
  application: ApplicationState,
  candidate: CandidateState,
  job: JobState,
}

interface AssessmentConsentProps {

}

type AssessmentConsentMergedProps = MapStateToProps & AssessmentConsentProps;

export const AssessmentConsent = (props: AssessmentConsentMergedProps) => {

  const { application, job, candidate } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId, scheduleId } = queryParams;
  const applicationData = application.results;
  const jobDetail = job.results;
  const { candidateData } = candidate.results;

  const [assessmentUrl, setAssessmentUrl] = useState<string>('');

  useEffect(() => {
    boundGetCandidateInfo();
  },[])

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() })
  }, [jobDetail, jobId]);

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    // Page will emit page load event once both pros are available but
    // will not emit new event on props change once it has emitted pageload event previously
    jobDetail && applicationData && candidateData && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData]);

  useEffect(() => {
    return () => {
      //reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    }
  },[])

  useEffect(() => {
    if (candidateData && jobDetail) {
      const assessmentType = jobDetail?.assessmentType;
      if (assessmentType && candidateData.assessmentsTaken) {
        setAssessmentUrl(candidateData.assessmentsTaken[assessmentType]?.assessmentUrl);
      }
    }
  }, [candidateData, jobDetail]);

  const diversityAndDisabilityText = t("BB-assessment-consent-diversity-and-disability-accommodation-notice-text", "At Amazon, we celebrate the diversity of our workforce and the diverse ways we work. If you have a disability and require an accommodation during the hiring process, including the assessment, please visit <a href='https://www.amazon.jobs/en/disability/us' target='_blank' rel='noopener noreferrer'> Amazon Accommodation.</a>");
  const assessmentPaceAndSectionHeadingText = t("BB-assessment-consent-accommodation-assessment-pace-and-sections-info-list-header-text", "<b>No part of the assessment is timed</b> â€“ you can work at your own pace. All sections of this assessment:");
  return (
    <Col gridGap={15}>
      <Col gridGap={10}>
        <H4>
          {t("BB-assessment-consent-assessment-heading-text", "Assessment")}
        </H4>
        <Text fontSize="T200">
          {t("BB-assessment-consent-excited-in-candidate-interest-text", "We are so excited in your interest to join Amazon!")}
        </Text>
        <ul>
          <li>
            <Text fontSize="T200">
              {t("BB-assessment-consent-purpose-of-assessment-text", "This assessment will help you understand the associate role and help us get to know you.")}
            </Text>
          </li>
        </ul>
      </Col>
      <Col gridGap={10}>
        <H4>
          {t("BB-assessment-consent-accommodation-heading-text", "Accommodations")}
        </H4>
        <Text fontSize="T200">
          <InnerHTML className="diversityAndDisabilityText" html={diversityAndDisabilityText}/>
        </Text>
        <Text fontSize="T200">
          <InnerHTML html={assessmentPaceAndSectionHeadingText} className="assessmentPaceAndSectionHeadingText"/>
        </Text>
        <ul>
          <li>
            <Text fontSize="T200">
              {t("BB-assessment-consent-accommodation-assessment-pace-and-sections-info-item1-text", "Are optimized for the following screen readers: VoiceOver (Mac) and NVDA (Windows) on Firefox, Chrome, and Edge (Windows).")}
            </Text>
          </li>
          <li>
            <Text fontSize="T200">
              {t("BB-assessment-consent-accommodation-assessment-pace-and-sections-info-item2-text", "Can be accessed via mouse or keyboard.")}
            </Text>
          </li>
        </ul>
        <Col
          padding={{ top: "S300" }}
        >
          <Button
            variant={ButtonVariant.Primary}
            onClick={() => assessmentUrl && applicationData && jobDetail && onAssessmentStart(assessmentUrl, applicationData, jobDetail)}
            disabled={!(assessmentUrl && applicationData && jobDetail)}
          >
            {t("BB-assessment-consent-begin-assessment-button-text", "Begin Assessment")}
          </Button>
        </Col>
      </Col>

    </Col>
  );
};

const mapStateToProps = ( state: MapStateToProps ) => {
  return state;
};

export default connect(mapStateToProps)(AssessmentConsent);
