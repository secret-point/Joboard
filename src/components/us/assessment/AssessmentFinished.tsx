import React, { useEffect } from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { useLocation } from "react-router-dom";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import queryString from "query-string";
import { JobState } from "../../../reducers/job.reducer";
import { ApplicationState } from "../../../reducers/application.reducer";
import { connect } from "react-redux";
import { checkAndBoundGetApplication, getLocale, routeToAppPageWithPath } from "../../../utils/helper";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { addMetricForPageLoad, postAdobeMetrics } from "../../../actions/AdobeActions/adobeActions";
import { QueryParamItem } from "../../../utils/types/common";
import { QUERY_PARAMETER_NAME, WORKFLOW_STEP_NAME } from "../../../utils/enums/common";
import { PAGE_ROUTES } from "../../pageRoutes";
import { onCompleteTaskHelper } from "../../../actions/WorkflowActions/workflowActions";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { METRIC_NAME } from "../../../constants/adobe-analytics";

interface MapStateToProps {
  job: JobState,
  application: ApplicationState,
  candidate: CandidateState,
}

export const AssessmentFinished = (props: MapStateToProps) => {
  const { job, application, candidate } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId } = queryParams;
  const jobDetail = job.results;
  const applicationData = application.results;
  const jobId = applicationData?.jobScheduleSelected.jobId;
  const scheduleId = applicationData?.jobScheduleSelected.scheduleId;
  const candidateData = candidate.results.candidateData;

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    boundGetCandidateInfo();
  },[]);

  useEffect(() => {
    // Page will emit page load event once both pros are available but
    // will not emit new event on props change once it has emitted pageload event previously
    jobDetail && applicationData && candidateData && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData, pageName]);

  useEffect(() => {
    return () => {
      //reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    }
  },[pageName])

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() }, () => {
      const queryParamItems: QueryParamItem[] = [
        {
          paramName: QUERY_PARAMETER_NAME.JOB_ID,
          paramValue: jobId
        }
      ];
      //Add schedule Id if exist in application
      if (scheduleId) {
        queryParamItems.push({
          paramValue: scheduleId,
          paramName: QUERY_PARAMETER_NAME.SCHEDULE_ID
        });
      }
      //force route to the same page to append query params ( jobId and schedule Id)
      routeToAppPageWithPath(PAGE_ROUTES.ASSESSMENT_FINISHED, queryParamItems);
      postAdobeMetrics({name: METRIC_NAME.ASSESSMENT_FINISHED});
      //call workflow service to update step
      applicationData && onCompleteTaskHelper(applicationData, false, undefined, WORKFLOW_STEP_NAME.ASSESSMENT_CONSENT);
    });
  }, [applicationData, jobDetail, jobId, scheduleId]);

  return (
    <Col minHeight="40vh"/>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(AssessmentFinished);
