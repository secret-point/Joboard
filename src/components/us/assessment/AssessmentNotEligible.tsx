import React, { useEffect } from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text, H4 } from "@amzn/stencil-react-components/text";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { checkAndBoundGetApplication, getLocale, goToCandidateDashboard } from "../../../utils/helper";
import { connect } from "react-redux";
import { ApplicationState } from "../../../reducers/application.reducer";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { useLocation } from "react-router-dom";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import queryString from "query-string";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { JobState } from "../../../reducers/job.reducer";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { translate as t } from "../../../utils/translator";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { ScheduleState } from "../../../reducers/schedule.reducer";

interface MapStateToProps {
  application: ApplicationState;
  candidate: CandidateState;
  job: JobState;
  schedule: ScheduleState;
}

interface AssessmentNotEligibleProps {

}

type AssessmentNotEligibleMergedProps = MapStateToProps & AssessmentNotEligibleProps;

export const AssessmentNotEligible = (props: AssessmentNotEligibleMergedProps) => {

  const { application, job, candidate } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId } = queryParams;
  const applicationData = application.results;
  const jobDetail = job.results;
  const { candidateData } = candidate.results;

  useEffect(() => {
    boundGetCandidateInfo();
  }, []);

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  // Don't refresh data if id is not changing
  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobDetail, jobId]);

  useEffect(() => {
    // Page will emit page load event once both pros are available but
    // will not emit new event on props change once it has emitted pageload event previously
    jobDetail && applicationData && candidateData && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData]);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, []);

  return (
    <Col gridGap={15}>
      <H4>
        {t("BB-assessment-not-eligible-heading-text", "{name}, your interest with Amazon means a lot to us.", { name: candidateData?.firstName || "" })}
      </H4>
      <Text fontSize="T200">
        {t("BB-assessment-not-eligible-thank-you-and-assessment-not-eligible-notice-text", "Thank you so much for giving us the opportunity to consider you for this position. Sadly, we will not be moving forward with your candidacy at this time.")}
      </Text>
      <Text fontSize="T200">
        {t("BB-assessment-not-eligible-reconsideration-notice-text", "To be reconsidered for this role at another time, you may take the assessment again in three months.")}
      </Text>
      <Col padding={{ top: "S300" }}>
        <Button
          variant={ButtonVariant.Primary}
          onClick={goToCandidateDashboard}
        >
          {t("BB-assessment-not-eligible-back-to-dashboard-button-text", "Return to dashboard")}
        </Button>
      </Col>
    </Col>
  );
};

const mapStateToProps = ( state: MapStateToProps ) => {
  return state;
};

export default connect(mapStateToProps)(AssessmentNotEligible);
