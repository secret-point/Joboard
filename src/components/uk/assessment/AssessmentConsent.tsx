import React, { useEffect } from "react";
import { ButtonVariant } from "@amzn/stencil-react-components/button";
import { FlyoutContent, WithFlyout } from "@amzn/stencil-react-components/flyout";
import { Col, } from "@amzn/stencil-react-components/layout";
import { H1, H3, Text } from "@amzn/stencil-react-components/text";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { JobState } from "../../../reducers/job.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { getLocale } from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import DebouncedButton from "../../common/DebouncedButton";
import { Link } from "@amzn/stencil-react-components/link";
import StepHeader from "../../common/StepHeader";
import { ApplicationStepListUK } from "../../../utils/constants/common";
import VideoContainer from "../../common/VideoContainer";
import { Expander } from "@amzn/stencil-react-components/expander";
import { onCompleteTaskHelper } from "../../../actions/WorkflowActions/workflowActions";
import { ApplicationState } from "../../../reducers/application.reducer";
import { APPLICATION_STEPS as STEPS } from "../../../utils/enums/common";
import { getStepsByTitle } from "../../../helpers/steps-helper";
import { AssessmentState } from "../../../reducers/assessment.reducer";
import { boundGetAssessmentElegibilitySuccess } from "../../../actions/AssessmentActions/boundAssessmentActions";
import { CandidateState } from "../../../reducers/candidate.reducer";

interface MapStateToProps {
  job: JobState;
  schedule: ScheduleState;
  application: ApplicationState;
  assessment: AssessmentState;
  candidate: CandidateState;
}

interface RenderFlyoutFunctionParams {
  close: () => void;
}

export const AssessmentConsent = (props: MapStateToProps) => {
  const { job, schedule, application, assessment } = props;
  const { search, pathname } = useLocation();
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { jobId, scheduleId } = queryParams;
  const jobDetail = job.results;
  const { scheduleDetail } = schedule.results;
  const pageName = getPageNameFromPath(pathname);
  const applicationData = application.results;
  const headerStep = getStepsByTitle(ApplicationStepListUK, STEPS.COMPLETE_AN_ASSESSMENT)[0]; 
  const { assessmentElegibility } = assessment.results;

  // Don't refetch data if id is not changing
  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobId]);

  useEffect(() => {
    scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    });
  }, [scheduleId]);

  useEffect(() => {
    if (jobDetail && ((scheduleId && scheduleDetail) || (!scheduleId))) {
      addMetricForPageLoad(pageName);
    }

  }, [jobDetail, scheduleDetail]);

  useEffect(() => {
    // Setting the assessmentElegibility to true to prevent subsequent request during this session
    assessmentElegibility === null && boundGetAssessmentElegibilitySuccess({
      assessmentElegibility: true });
  }, []);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, []);

  const handleConsentToAssessment = () => {
    applicationData && onCompleteTaskHelper(applicationData);
  };

  const renderAssessmentPolicyFlyout = ({ close }: RenderFlyoutFunctionParams) => (
    <FlyoutContent
      titleText={t("BB-AssessmentConsentPage-assessment-policy-flyout-title", "Assessment Policy")}
      onCloseButtonClick={close}
    >
      <Col gridGap={15}>
        <H3 fontWeight="bold" fontSize="T400">
          {t("BB-AssessmentConsentPage-assessment-policy-flyout-acknowledgement-title", "Acknowledgement")}
        </H3>
        <Text>
          {t("BB-AssessmentConsentPage-assessment-policy-flyout-acknowledgement-intro", "By taking the assessment, I acknowledge that:")}
        </Text>
        <ul>
          <li>
            <Text>
              {t("BB-AssessmentConsentPage-assessment-policy-flyout-acknowledgement-list-item-1", "Responses to all assessments in this application process are my own.")}
            </Text>
          </li>

          <li>
            <Text>
              {t("BB-AssessmentConsentPage-assessment-policy-flyout-acknowledgement-list-item-2", "I did not consult any other person or use any additional resources while completing the assessments.")}
            </Text>
          </li>
          <li>
            <Text>
              {t("BB-AssessmentConsentPage-assessment-policy-flyout-acknowledgement-list-item-3", "I understand that I may be asked to justify my answers and that any misrepresentation will result in rejection of my application.")}
            </Text>
          </li>

        </ul>
      </Col>
    </FlyoutContent>
  );

  return (
    <>
      <Col gridGap="S300" padding="0">
        <StepHeader jobTitle={jobDetail?.jobTitle || ""} step={headerStep} steps={ApplicationStepListUK} />

        <H1 fontSize="T400" fontWeight="regular">
          {t("BB-AssessmentConsentPage-title-text", "Assessment")}
        </H1>

        <Text fontSize="T200">
          {t("BB-AssessmentConsentPage-intro", "Evaluate your fit at Amazon.")}
        </Text>

        <VideoContainer src="https://static-assets.associate.amazondelivers.jobs/media-assets/Blackbird_Assessment_Intro_20200928.mp4" id="assessment-video" poster="https://m.media-amazon.com/images/G/01/HVH-CandidateApplication/AssessmentPoster.jpg" />

        <Expander titleText={t("B-AssessmentConsentPage-conditions-title-text-expanderTitle", "Read assessment instructions")}>

          <Text>
            {t("BB-AssessmentConsentPage-conditions-title-text-expanderContent-intro", "You are about to complete an assessment that is part of the application process.")}
          </Text>

          <ul>
            <li>
              <Text>
                {t("BB-AssessmentConsentPage-conditions-title-text-expanderContent-list-item-1", "We recommend that you find a quiet place to avoid distractions.")}
              </Text>
            </li>
            <li>
              <Text>
                {t("BB-AssessmentConsentPage-conditions-title-text-expanderContent-list-item-2", "After you finish the assessment, you will be returned to your application process.")}
              </Text>
            </li>
            <li>
              <Text>
                {t("BB-AssessmentConsentPage-conditions-title-text-expanderContent-list-item-3", "Our inclusive culture empowers Amazonians to deliver the best results for our customers. If you have a disability and need an adjustment during the application and hiring process, including support for the interview or onboarding process, please contact the Applicant-Candidate Accommodation Team (ACAT), Monday through Friday from 7:00 am GMT - 4:00 pm GMT . If calling directly from the United Kingdom, please dial +44 800 086 9884. If calling from Ireland, please dial +353 1800 851 489. You may also contact us if you might need an adjustment in your new role and would like to initiate a request prior to starting your Day 1.")}
              </Text>
            </li>
          </ul>

        </Expander>
        <Text margin={{ bottom: "S400" }} fontSize="T200">
          {t("BB-AssessmentConsentPage-title-text-introduction", "By beginning the assessment, you agree to the")}
          <WithFlyout renderFlyout={renderAssessmentPolicyFlyout}>
            {({ open }) => (
              <Link
                onClick={() => open()}
                margin={{
                  left: "S100"
                }}
              >
                {t("BB-AssessmentConsentPage-assessment-policy-flyout-buttonLabel", "Assessment Policy")}
              </Link>
            )}
          </WithFlyout>
        </Text>
        <Col gridGap={8}>
          <DebouncedButton
            variant={ButtonVariant.Primary}
            style={{ width: "100%" }}
            onClick={() => {
              handleConsentToAssessment();
            }}
          >
            {t("BB-AssessmentConsentPage-start-assessment-button", "Start Assessment")}
          </DebouncedButton>
        </Col>
      </Col>
    </>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(AssessmentConsent);
