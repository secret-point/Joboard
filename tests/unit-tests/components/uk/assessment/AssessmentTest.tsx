import React from "react";
import { shallow } from "enzyme";
import { Assessment } from "../../../../../src/components/uk/assessment/Assessment";
import {
  TEST_APPLICATION_ID,
  TEST_APPLICATION_STATE,
  TEST_CANDIDATE_STATE,
  TEST_JOB_ID,
  TEST_JOB_STATE,
  TEST_SCHEDULE_ID,
  TEST_SCHEDULE_STATE
} from "../../../../test-utils/test-data";
import { useLocation } from "react-router-dom";

describe("Assessment", () => {

  const mockLocation = {
    pathname: "/job-opportunities",
    search: `?jobId=${TEST_JOB_ID}&applicationId=${TEST_APPLICATION_ID}&scheduleId=${TEST_SCHEDULE_ID}`,
    hash: "",
    state: null
  };
  const mockUseLocation = useLocation as jest.Mock;
  mockUseLocation.mockReturnValue(mockLocation);

  it("should match snapshot", () => {
    const shallowWrapper = shallow(
      <Assessment
        job={TEST_JOB_STATE}
        application={TEST_APPLICATION_STATE}
        schedule={TEST_SCHEDULE_STATE}
        candidate={TEST_CANDIDATE_STATE}
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });
});
