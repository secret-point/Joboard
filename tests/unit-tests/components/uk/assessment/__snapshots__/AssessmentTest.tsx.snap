// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assessment should match snapshot 1`] = `
<Fragment>
  <Col
    gridGap="S300"
    padding="0"
  >
    <StepHeader
      jobTitle=""
      step={
        Object {
          "title": "Complete an assessment (~15 min)",
          "titleTranslationKey": "BB-ApplicationSteps-complete-assessment-text",
        }
      }
      steps={
        Array [
          Object {
            "title": "Complete an assessment (~15 min)",
            "titleTranslationKey": "BB-ApplicationSteps-complete-assessment-text",
          },
          Object {
            "title": "Select Job",
            "titleTranslationKey": "BB-ApplicationSteps-select-job-text",
          },
          Object {
            "title": "Enter required information",
            "titleTranslationKey": "BB-ApplicationSteps-enter-required-information-text",
          },
          Object {
            "title": "Schedule pre-hire appointment",
            "titleTranslationKey": "BB-ApplicationSteps-schedule-preHire-appt-text",
          },
        ]
      }
    />
  </Col>
</Fragment>
`;
